File: Structural/Decorator/src/Product.php
<?php

namespace Structural\Decorator;

class Product implements HasPrice
{
    private $id;
    private $name;
    private $price;

    public function __construct(string $name, float $price)
    {
        $this->id = uniqid();
        $this->name = $name;
        $this->price = $price;
    }

    public function getId()
    {
        return $this->id;
    }

    public function getName()
    {
        return $this->name;
    }

    public function getPrice(): float
    {
        return $this->price;
    }
}

File: oop-pattern/ch07/example/app/src/main/java/study/facade/Product.java
package study.facade;

public class Product {

  int productId;
  String name;
  int price;

  public Product() {
  }

  public Product(int productId, String name, int price) {
    this.productId = productId;
    this.name = name;
    this.price = price;
  }

  public int getProductId() {
    return productId;
  }

  public String getName() {
    return name;
  }

  public int getPrice() {
    return price;
  }

  @Override
  public String toString() {
    return "Product{" +
        "productId=" + productId +
        ", name='" + name + '\'' +
        ", price=" + price +
        '}';
  }
}
